### PR TITLE
sign tx with hash functionality for hw provider

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Updated the `erdjs`'s hardware wallet signing to the latest version of `hw-app-elrond`.  
Now it checks the Ledger's App version when signing a transaction and calls the library's method the way it should by also providing a `boolean` parameter

### CODE REVIEW

File: erdjs/src/dapp/hwProvider.ts [hunk 1]
```diff
@@ -90,15 +94,26 @@ export class HWProvider implements IDappProvider {
...
+        let signWithHash = await this.shouldSignWithHash();
+        const sig = await this.hwApp.signTransaction(transaction.serializeForSigning(new Address(address)), signWithHash);
         transaction.signature = new Signature(sig);
 
         await transaction.send(this.provider);
 
         return transaction;
     }
 
+    private async shouldSignWithHash(): Promise<boolean> {
```
Comment (andreibancioiu) [inline erdjs/src/dapp/hwProvider.ts:106]: :+1:

File: erdjs/src/dapp/hwProvider.ts [hunk 2]
```diff
@@ -90,15 +94,26 @@ export class HWProvider implements IDappProvider {
...
+        if (!this.hwApp) {
+            throw new Error("HWApp not initialised, call init() first");
```
Comment (andreibancioiu) [inline erdjs/src/dapp/hwProvider.ts:108]: You can also define specific errors in `errors.ts` - not necessarily in this case, but in general. Fine as it is.

File: erdjs/src/dapp/hwProvider.ts [hunk 3]
```diff
@@ -90,15 +94,26 @@ export class HWProvider implements IDappProvider {
...
+        }
+
+        const config = await this.hwApp.getAppConfiguration();
```
Comment (andreibancioiu) [inline erdjs/src/dapp/hwProvider.ts:111]: `await this.getAppVersion()` (new function) - optional.

**Review by andreibancioiu — DISMISSED** [review]

**Comment by andreibancioiu** [comment]
Please add a description to the PR - when making releases, we'll point to the this PR in the changelog file.

**Review by ccorcoveanu — CHANGES_REQUESTED** [review]

File: erdjs/src/dapp/common.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+export function isLedgerVersionForSigningUsingHash(version: string) : boolean {
+    let i, diff;
+    const regExStrip0 = /(\.0+)+$/;
+    const segmentsA = version.replace(regExStrip0, '').split('.');
+    const segmentsB = LEDGER_TX_HASH_SIGN_MIN_VERSION.replace(regExStrip0, '').split('.');
+    const l = Math.min(segmentsA.length, segmentsB.length);
+
+    for (i = 0; i < l; i++) {
+        diff = parseInt(segmentsA[i], 10) - parseInt(segmentsB[i], 10);
+        if (diff < 0) {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply erdjs/src/dapp/common.ts]: done

File: erdjs/src/dapp/common.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import {LEDGER_TX_HASH_SIGN_MIN_VERSION} from './constants';
+
+export function isLedgerVersionForSigningUsingHash(version: string) : boolean {
+    let i, diff;
+    const regExStrip0 = /(\.0+)+$/;
+    const segmentsA = version.replace(regExStrip0, '').split('.');
+    const segmentsB = LEDGER_TX_HASH_SIGN_MIN_VERSION.replace(regExStrip0, '').split('.');
+    const l = Math.min(segmentsA.length, segmentsB.length);
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply erdjs/src/dapp/common.ts]: done

File: erdjs/src/dapp/common.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import {LEDGER_TX_HASH_SIGN_MIN_VERSION} from './constants';
+
+export function isLedgerVersionForSigningUsingHash(version: string) : boolean {
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply erdjs/src/dapp/common.ts]: done

**Review by ccorcoveanu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**